app/config.go

```diff
@@ -526,7 +526,18 @@ var configurers = []configurer{
 	// PORT is the local port the AuthN server listens to. The default is taken from AUTHN_URL, but
 	// may be different for port mapping scenarios as with containers and load balancers.
 	func(c *Config) error {
-		defaultPort, err := strconv.Atoi(c.AuthNURL.Port())
+		var defaultPort int
+
+		if p := c.AuthNURL.Port(); p != "" {
+			defaultPort, _ = strconv.Atoi(p)
+		} else {
+			switch c.AuthNURL.Scheme {
```

Comment:
It seems like this would be needed for scenarios where authn URL is configured with a default port - far as I can tell you would need to explicitly configure port for those to work which is weird.